### PR TITLE
docs(mantine): add useSelect doc

### PR DESCRIPTION
**Pull request by salihozdemir**
Added the Mantine useSelect hook doc.

### Test plan (required)

Demonstrate the code is solid. If not, please add `WIP:` in its title.

### Closing issues

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: documentation/docs/api-reference/mantine/hooks/useSelect.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+[Refer to Mantine `Select` component documentation for detailed info for `options`. &#8594](https://mantine.dev/core/select/)
+
+### `defaultValue`
+
+```tsx
+const { selectProps } = useSelect({
+    resource: "categories",
+    // highlight-next-line
+    defaultValue: "1",
```
Comment (omeraplak) [inline documentation/docs/api-reference/mantine/hooks/useSelect.md:125]:
```suggestion
    defaultValue: 1,
```

File: documentation/docs/api-reference/mantine/hooks/useSelect.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+});
+```
+
+Adds extra `options` to `<Select>` component. It uses `useMany` so `defaultValue` can be an array of strings like follows.
+
+```ts
+defaultValue: ["1", "2"],
```
Comment (omeraplak) [inline documentation/docs/api-reference/mantine/hooks/useSelect.md:132]:
```suggestion
defaultValue: [1, 2],
```

File: documentation/docs/api-reference/mantine/hooks/useSelect.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+};
+```
+
+`useSelect` hook allows you to manage an Mantine [Select](https://mantine.dev/core/select/) component when records in a resource needs to be used as select options.
+
+## Usage
+
+We'll demonstrate how to get data at `/categories` endpoint from `https://api.fake-rest.refine.dev` REST API and use it as select options. Also, you'll see how to use `Select` component with `useSelect` hook and `useForm` hook.
+
+```tsx live url=http://localhost:3000/posts/create previewHeight=420px
```
Comment (alicanerdurmaz) [inline documentation/docs/api-reference/mantine/hooks/useSelect.md:36]: `url=/posts/create` shouldn't it be like this?

Comment (salihozdemir) [reply documentation/docs/api-reference/mantine/hooks/useSelect.md:36]: It is just a dump URL that we want to appear on the live preview window. We set `localhost:3000` to all of other examples. So, I think it's OK 😅

**Pull request merged**